clientd3d/d3drender_objects.h

```diff
@@ -13,6 +13,153 @@
 #ifndef _D3DRENDEROBJECTS_H
 #define _D3DRENDEROBJECTS_H
 
-void D3DRenderObjects(room_type* room, Draw3DParams* params, room_contents_node* pRNode);
+struct PlayerViewParams
+{
+    // Player information
+    player_info* player;
+
+    // Screen dimensions
+    int screenWidth;
+    int screenHeight;
+
+    // Viewport dimensions
+    int viewportWidth;
+    int viewportHeight;
+
+    // D3D viewport rectangle
+    RECT d3dRect;
+
+    PlayerViewParams(
+        player_info* playerParam,
+        int screenWidthParam,
+        int screenHeightParam,
+        int viewportWidthParam,
+        int viewportHeightParam,
+        RECT d3dRectParam)
+        : player(playerParam),
+          screenWidth(screenWidthParam),
+          screenHeight(screenHeightParam),
+          viewportWidth(viewportWidthParam),
+          viewportHeight(viewportHeightParam),
+          d3dRect(d3dRectParam)
+    {}
+};
+
+struct FontTextureParams
+{
+    font_3d* font;
+    Color* basePalette;
+    int smallTextureSize;
+    int* sectorDepths;
```

Comment:
sector depths definitely doesn't belong here; it has nothing to do with the font texture.